.github/workflows/kind.yml

```diff
@@ -1,51 +1,95 @@
 name: E2E
-
 on:
   push:
     branches:
       - master
-    paths-ignore:
-      - "docs/**"
+      - main
   pull_request:
     types: [opened, synchronize, reopened]
-    paths-ignore:
-      - "docs/**"
 
 jobs:
+  gates:
+    # The gates job is intended to derive all necessary information for branching logic further down the graph.
+    runs-on: ubuntu-latest
+    outputs:
+      # See jobs.<job_id>.outputs reference for details on passing variables between jobs:
+      # https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idoutputs
+      commitmsg: ${{ steps.getcommit.outputs.commitmsg }}
+      fileChangesToCoreArtifacts: ${{ steps.filter.outputs.fileChangesToCoreArtifacts }}
+    steps:
+    - uses: actions/checkout@main
+      with:
+        fetch-depth: 10
+    - name: get-commit-msg
+      id: getcommit
+      run: |
+        echo ::set-output name=commitmsg::$(git log --format=%B -n 1 ${{ github.event.after }})
+    - uses: dorny/paths-filter@v2
+      # Reference documentation for this action: https://github.com/dorny/paths-filter#examples 
+      id: filter
+      with:
+        # Here we test for file changes with the filter. While at present the "fileChangesToCoreArtifacts" variable
+        # is only testing for changes outside of the documentation, this does in principle give us fine-grained
+        # control should we wish to exclude other supporting folders from triggering the larger test runs.
+        # 
+        # Unfortunately, it is, at time of writing, not possible to do this work with Github Action's path
+        # exclusions logic, as there is no way to set a job to report as successful without running it, and
+        # the paths logic will skip the job altogether. This means that, without the control gained here,
+        # any job that we wish to set as "Required" in for merge checks must be run whether it's relevant 
+        # to the changes or not. See the following discussion: 
+        # https://github.community/t/feature-request-conditional-required-checks/16761/13
+        filters: |
+          fileChangesToCoreArtifacts:
+            - '!docs/**'
   binary:
     runs-on: ubuntu-20.04
+    needs: gates
     container:
       image: flanksource/build-tools:v0.12.0
+    env: 
+      FILE_CHANGES_TO_CORE_ARTIFACTS: ${{ needs.gates.outputs.fileChangesToCoreArtifacts == 'true' }}
     steps:
       # Get values for cache paths to be used in later steps
       - id: go-cache-paths
+        if: ${{ env.FILE_CHANGES_TO_CORE_ARTIFACTS == 'true' }}
```

Comment:
Is it possible to have a step that essentially is `if ${{ env.FILE_CHANGES_TO_CORE_ARTIFACTS == 'false' }}` then succeed without running the other steps?

Reply:
Not that's I've been able to find. My first impulse was to try early return but they don't seem to operate that way (I could be wrong) and it seems without further research I think the idea is that that they're independent without explicitly stating deps.

Reply:
Ok. That is slightly annoying on GA's part...